samples/channels_buffered.go

```diff
@@ -0,0 +1,26 @@
+// channels_buffered
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+func channelWrite(ch chan<- int, value int) {
+	fmt.Println("started channel write", value)
+	ch <- value
+	fmt.Println("finished channel write", value)
+}
+
+func main() {
+	ch := make(chan int, 5)
```

Comment:
I wonder if it might be more demonstrative if the channel buffer size is less than the loop size, so you can see the first _n_ sends go through without blocking, but the subsequent ones block until a receive is done?

Or perhaps if you present this code during the presentation, you could start with this, but then show the class what happens when you change the buffer size to 2, for example?

Reply:
I will remember to explain this bit in the demo. Hope that's okay. ? Should I add a comment to the code to remind myself ?

Reply:
It is possible to add presenter notes to the slides (see https://godoc.org/golang.org/x/tools/present and search for [presenter notes]) but we haven't used them at all in this slide deck and it relies on having a second screen. I assume we're probably mirroring the screen to the projector so we can see the slides ourselves without turning around, so presenter notes probably wont work.

You could add a comment at this line:
```
// What happens if the channel buffer size is 2?
```

This may be enough of a prompt to you and also gets the more adventurous students to experiment.